redash/models/__init__.py

```diff
@@ -362,9 +362,21 @@ def get_latest(cls, data_source, query, max_age=0):
         query_hash = gen_query_hash(query)
 
         if max_age == -1:
-            query = cls.query.filter(
-                cls.query_hash == query_hash, cls.data_source == data_source
-            )
+            if settings.QUERY_RESULTS_EXPIRED_TTL_ENABLED:
+                query = cls.query.filter(
+                    cls.query_hash == query_hash,
+                    cls.data_source == data_source,
+                    (
+                        db.func.timezone("utc", cls.retrieved_at) +
+                        datetime.timedelta(
+                            seconds=settings.QUERY_RESULTS_EXPIRED_TTL
+                        ) >= db.func.timezone("utc", db.func.now())
+                    )
+                )
```

Comment:
Can we merge this part of the logic with "max_age != -1"? They are almost the same.

Reply:
You can change it.

Reply:
> Can we merge this part of the logic with "max_age != -1"? They are almost the same.

When max != -1, it is to actively refresh or the query does not get the corresponding result.

Reply:
> Can we merge this part of the logic with "max_age != -1"? They are almost the same.

In redash, `max_age` in `0 or -1` two types.

Reply:
@junnplus  You can review this code, and then i want to show your logic for my mind.

Reply:
> Can we merge this part of the logic with "max_age != -1"? They are almost the same.

What I mean is that we can reset the value of max_age before `if max_age == -1:`.
```python
        if max_age == -1 and settings.QUERY_RESULTS_EXPIRED_TTL_ENABLED:
            max_age = settings.QUERY_RESULTS_EXPIRED_TTL

        if max_age:
            ...
```

Reply:
> > Can we merge this part of the logic with "max_age != -1"? They are almost the same.
> 
> What I mean is that we can reset the value of max_age before `if max_age == -1:`.
> 
> ```python
>         if max_age == -1 and settings.QUERY_RESULTS_EXPIRED_TTL_ENABLED:
>             max_age = settings.QUERY_RESULTS_EXPIRED_TTL
> 
>         if max_age:
>             ...
> ```

Got it, we can change to this logic.